src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery.Test/ScenarioTests/A2A/AsrA2ATests.ps1

```diff
@@ -0,0 +1,673 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+########################## Site Recovery Tests #############################
+
+##Default Value ##
+$vaultName = "A2APowershellTest"
+$vaultLocation = "centraluseuap"
+$vaultRg = "A2APowershellTestRg"
+$primaryLocation = "eastasia" 
+$recoveryLocation = "southeastasia"
+$primaryFabricName = "a2aPrimaryFabric3"
+$recoveryFabricName = "a2aRecoveryFabric3"
+                         
+$PrimaryAzureNetworkId = "/subscriptions/7c943c1b-5122-4097-90c8-861411bdd574/resourceGroups/powershellTest/providers/Microsoft.Network/virtualNetworks/powershellTest-vnet"
```

Comment:
Azure Site Recovery works on existing resources .And it needs N number of preexisting resources setup recovery . For Azure site Recovery  rerecording thing will not be easy as Enterprise infra will be non-existing at time of reRecording. ASR cmdlet runs in context of vault and subscription which already is imported from vaultCredentials file.  If someone is rerecording he should have prior knowledge of creating infra to reRun and he can pass infra details at starting of the file.
If we put the logic of infra creation in test it will be pretty complex.

Can discuss if there is easy to make this independent.